src/Aspire.Hosting.Azure/Provisioning/Internal/DefaultTokenCredentialProvider.cs

```diff
@@ -25,7 +25,6 @@ public DefaultTokenCredentialProvider(ILogger<DefaultTokenCredentialProvider> lo
             "AzureCli" => new AzureCliCredential(),
             "AzurePowerShell" => new AzurePowerShellCredential(),
             "VisualStudio" => new VisualStudioCredential(),
-            "VisualStudioCode" => new VisualStudioCodeCredential(),
```

Comment:
cc: @eerhardt @davidfowl wonder what your thoughts are around the breaking change here. I did it in this PR as the update of the Azure dependency marked this type (`VisualStudioCodeCredential`) as deprecated which broke our build. I've hence removed this as an option for now but of course that could be breaking in the scenario someone was using this, so let me know if you prefer me just suppress the warning until 10.

Reply:
The full error we were getting is:

> /home/runner/work/aspire/aspire/src/Aspire.Hosting.Azure/Provisioning/Internal/DefaultTokenCredentialProvider.cs(28,39): error CS0618: 'VisualStudioCodeCredential' is obsolete: 'This credential is deprecated because the VS Code Azure Account extension on which this credential relies has been deprecated. Consider using other dev-time credentials, such as VisualStudioCredential, AzureCliCredential, AzureDeveloperCliCredential, AzurePowerShellCredential. See the Azure Account extension deprecation notice here: https://github.com/microsoft/vscode-azure-account/issues/964.' [/home/runner/work/aspire/aspire/src/Aspire.Hosting.Azure/Aspire.Hosting.Azure.csproj]

Reply:
This looks good to me.

Reply:
Yea this is fine. We should mark it a breaking change.

Reply:
FYI: @IEvangelist as this is a breaking change.

Reply:
I looked at https://learn.microsoft.com/en-us/dotnet/aspire/azure/local-provisioning#azure-provisioning-credential-store and we never documented `VisualStudioCode` as a possible value.